lib/resin.coffee

```diff
@@ -30,6 +30,15 @@ RESIN_SDK_SHARED_OPTIONS = 'RESIN_SDK_SHARED_OPTIONS'
 RESIN_SDK_HAS_USED_SHARED_OPTIONS = 'RESIN_SDK_HAS_USED_SHARED_OPTIONS'
 RESIN_SDK_HAS_SET_SHARED_OPTIONS = 'RESIN_SDK_HAS_SET_SHARED_OPTIONS'
 
+# Use window (web)/self (web worker)/global (node) as appropriate
+globalEnv = if typeof window != 'undefined'
+	window
+else if typeof self != 'undefined'
+	self
+else if typeof global != 'undefined'
+	global
+else null
```

Comment:
How about returning an empty object instead of `null`? This way the code will continue to work even on that case. It could be harder to debug though.

Reply:
Yeah, I thought about that, but I think it's better if it explodes, which is why this explicitly avoids that. I have added a comment though, because you're right, this isn't clear.

This is only for the `sharedOptions` functions, and if you try to call those in a state where we can't guarantee that the options will be reliably shared (e.g. if you share them in an object here, they'll be shared for one install of this the SDK, but not _all_ installs of it, if you have npm conflicts etc) then imo it's better for it to make that clear, so you can handle it yourself properly.

Reply:
Oh, fair enough, I didn't have that much context before and didn't think of that that way.
PS: Liked the `explodes` part :boom: !